### PR TITLE
hash-cache-tool: Prints capitalization and num accounts when diffing files

### PR DESCRIPTION
**Pull request by brooksprumo**
#### Problem

The hash cache tool does not print the number of accounts nor capitalization when diffing files.

#### Summary of Changes

When extracting entries, also sum the lamports for capitalization. Then print out this info.

We'll be using this more when diffing the whole state too.

### CODE REVIEW

File: accounts-db/accounts-hash-cache-tool/src/main.rs [hunk 1]
```diff
@@ -465,12 +486,10 @@ fn get_cache_files_in(dir: impl AsRef<Path>) -> Result<Vec<CacheFileInfo>, io::E
...
 }
 
-/// Returns the entries in `file`
+/// Returns the entries in `file`, and the capitalization
 ///
 /// If there are multiple entries for a pubkey, only the latest is returned.
-fn extract_latest_entries_in(
-    file: impl AsRef<Path>,
-) -> Result<HashMap<Pubkey, (AccountHash, /* lamports */ u64)>, String> {
+fn extract_latest_entries_in(file: impl AsRef<Path>) -> Result<LatestEntriesInfo, String> {
```
Comment (jeffwashington) [inline accounts-db/accounts-hash-cache-tool/src/main.rs:492]: clippy must've made you add `LatestEntriesInfo`. I know how much you dislike types like this!

File: accounts-db/accounts-hash-cache-tool/src/main.rs [hunk 2]
```diff
@@ -181,11 +181,32 @@ fn do_inspect(file: impl AsRef<Path>, force: bool) -> Result<(), String> {
...
+    let LatestEntriesInfo {
+        latest_entries: entries2,
+        capitalization: capitalization2,
+    } = extract_latest_entries_in(&file2)
         .map_err(|err| format!("failed to extract entries from file 2: {err}"))?;
 
+    let num_accounts1 = entries1.len();
+    let num_accounts2 = entries2.len();
+    let num_accounts_width = {
+        let width1 = (num_accounts1 as f64).log10().ceil() as usize;
```
Comment (jeffwashington) [inline accounts-db/accounts-hash-cache-tool/src/main.rs:198]: i don't understand `log10`

File: accounts-db/accounts-hash-cache-tool/src/main.rs (hunk 1 above)
Comment (brooksprumo) [reply accounts-db/accounts-hash-cache-tool/src/main.rs:492]: Hah, clippy did indeed!

File: accounts-db/accounts-hash-cache-tool/src/main.rs (hunk 2 above)
Comment (brooksprumo) [reply accounts-db/accounts-hash-cache-tool/src/main.rs:198]: This is a log base-10. Getting the number of characters of the number, so I can pretty print the width and have them all be aligned.

**Review by jeffwashington — APPROVED** [review]
lgtm

**Review by HaoranYi — COMMENTED** [review]
lgtm.

**Pull request merged**